beacon/light/committee_chain.go

```diff
@@ -0,0 +1,555 @@
+				return nil, err
+			}
+			return update, nil
+		}),
+		syncCommitteeCache: lru.NewCache[uint64, syncCommittee](10),
+		db:                 db,
+		sigVerifier:        sigVerifier,
+		clock:              clock,
+		unixNano:           unixNano,
+		config:             config,
+		signerThreshold:    signerThreshold,
+		enforceTime:        enforceTime,
+		minimumUpdateScore: types.UpdateScore{
+			SignerCount:    uint32(signerThreshold),
+			SubPeriodIndex: params.SyncPeriodLength / 16,
+		},
+	}
+
+	// check validity constraints
+	if !s.updates.IsEmpty() {
+		if s.fixedRoots.IsEmpty() || s.updates.First < s.fixedRoots.First ||
+			s.updates.First >= s.fixedRoots.AfterLast {
+			log.Crit("Inconsistent database error: first update is not in the fixed roots range")
+		}
+		if s.committees.First > s.updates.First || s.committees.AfterLast <= s.updates.AfterLast {
+			log.Crit("Inconsistent database error: missing committees in update range")
+		}
+	}
+	if !s.committees.IsEmpty() {
+		if s.fixedRoots.IsEmpty() || s.committees.First < s.fixedRoots.First ||
+			s.committees.First >= s.fixedRoots.AfterLast {
+			log.Crit("Inconsistent database error: first committee is not in the fixed roots range")
+		}
+		if s.committees.AfterLast > s.fixedRoots.AfterLast && s.committees.AfterLast > s.updates.AfterLast+1 {
+			log.Crit("Inconsistent database error: last committee is neither in the fixed roots range nor proven by updates")
+		}
+		log.Trace("Sync committee chain loaded", "first period", s.committees.First, "last period", s.committees.AfterLast-1)
+	}
+	// roll back invalid updates (might be necessary if forks have been changed since last time)
+	var batch ethdb.Batch
+	for !s.updates.IsEmpty() {
+		if update := s.updates.get(s.updates.AfterLast - 1); update == nil || s.verifyUpdate(update) {
+			if update == nil {
+				log.Crit("Sync committee update missing", "period", s.updates.AfterLast-1)
+			}
+			break
+		}
+		if batch == nil {
+			batch = s.db.NewBatch()
+		}
+		s.rollback(batch, s.updates.AfterLast)
+	}
+	if batch != nil {
+		if err := batch.Write(); err != nil {
+			log.Error("Error writing batch into chain database", "error", err)
+		}
+	}
+	return s
+}
+
+// Reset resets the committee chain.
+func (s *CommitteeChain) Reset() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	batch := s.db.NewBatch()
+	s.rollback(batch, 0)
+	if err := batch.Write(); err != nil {
+		log.Error("Error writing batch into chain database", "error", err)
+	}
+}
+
+// AddFixedRoot sets a fixed committee root at the given period.
+// Note that the period where the first committee is added has to have a fixed
+// root which can either come from a CheckpointData or a trusted source.
+func (s *CommitteeChain) AddFixedRoot(period uint64, root common.Hash) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	batch := s.db.NewBatch()
+	oldRoot := s.getCommitteeRoot(period)
+	if !s.fixedRoots.CanExpand(period) {
+		if root != oldRoot {
```

Comment:
It's a bit confusing. 

Here, if the provided fixed root is not matched with the one we have, it's rejected.
But in line 203, we blindly accept the new root and rollback the chain.

I guess some comments/descriptions is necessary

Reply:
Added comments about the logic